src/module.c

```diff
@@ -1584,8 +1644,65 @@ int RM_ReplyWithEmptyArray(RedisModuleCtx *ctx) {
     return REDISMODULE_OK;
 }
 
+/* Reply to the client with an empty map.
+ *
+ * The function always returns REDISMODULE_OK. */
+int RM_ReplyWithEmptyMap(RedisModuleCtx *ctx) {
+    client *c = moduleGetReplyClient(ctx);
+    if (c == NULL) return REDISMODULE_OK;
+    addReply(c,shared.emptymap[c->resp]);
+    return REDISMODULE_OK;
+}
+
+/* Reply to the client with an empty set.
+ *
+ * The function always returns REDISMODULE_OK. */
+int RM_ReplyWithEmptySet(RedisModuleCtx *ctx) {
```

Comment:
What should be the behavior for an empty attribute?

Reply:
to be honest, i don't think we wanna have that API, or none of the "WithEmpty" APIs.
i suppose for strings it makes some sense, and for arrays it's too late to remove, but i don't think we wanna add more of these.

Reply:
furthermore, i remember we had some discussion about removing them, or was it about unifying the `RM_ReplySet<xxx>Length` APIs into one?
i can't find that discussion, but i do think we have too many APIs per collection type.

Reply:
ohh, i do see we deprecated REDISMODULE_POSTPONED_ARRAY_LEN in favor of REDISMODULE_POSTPONED_LEN, so i guess that's what i remember we discussed (to eliminate some repetition) 

still i think ReplyWithEmptyMap and ReplyWithEmptySet should be removed.